assets/js/header.js

```diff
@@ -0,0 +1,20 @@
+
+const headerElement = document.querySelector('.header');
+
+export function evaluateStickyHeader() {
+    if (window.scrollY > 50) {
+        makeHeaderSticky();
+    } else {
+        makeHeaderUnsticky();
+    }
+}
+
+function makeHeaderSticky() {
+    document.body.classList.add('--sticky');
```

Comment:
In the BEM naming convention, we use the `--something`-suffix as a “modifier”. We then use the resulting class in addition to the respective base-class. We would not use `<header class="header--variation">`, but `<header class="header header--variation">`.

Chances are that a sticky header is mostly like a normal header, but has a few modifications that override what the base-header provides. Most properties go with `.header`, only the changes go with `.header--variation`:

```sass
.header {
  // all the usual header stuff

  &--variation {
    // overrides for this (static) modification
  }
}
```

The kicker is that those “modifiers” stay the same while we are on a page; they don’t change dynamically. In the above example, we would not toggle `.header--variation` on and off with JavaScript.

We also have “states”, which we give through _additional_ classes. Those we do toggle via JavaScript. As a rule of thumb, if you can add a boolean-type prefix (“is”, “has”, “did”, etc) to something, and if it can change back and forth, it’s probably a state.

We don’t have to namespace states to the component they are attached to. For the sticky header, we would use something like `<header class="header is-sticky">`. It will always be a `.header`, but only `.is-sticky` every once in a while.

In the Sass, we must then add the ampersand (`&`) to the selector to avoid accidental nesting:

```sass
.header {
  // all the usual header-stuff

  &.is-sticky {
    // overrides for when this is sticky
  }
}
```

Following this pattern, you probably want to toggle a class of `has-sticky-header` on the body.